src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -80,6 +80,8 @@ public override bool IsImplicitlyDeclared
             get { return null; }
         }
 
+        internal virtual bool HasDefaultArgumentSyntax => false;
```

Comment:
Introducing an API like this feels wrong. A synthesized parameter never has syntax and cannot have it for default value as well